packages/trpc/server/routers/viewer/workflows/update.handler.ts

```diff
@@ -289,132 +293,206 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
   }
 
   // handle deleted and edited workflow steps
-  userWorkflow.steps.map(async (oldStep) => {
-    const foundStep = steps.find((s) => s.id === oldStep.id);
-    let newStep;
-
-    if (foundStep) {
-      const { senderName, ...rest } = {
-        ...foundStep,
-        numberVerificationPending: false,
-        sender: getSender({
-          action: foundStep.action,
-          sender: foundStep.sender || null,
-          senderName: foundStep.senderName,
-        }),
-      };
-      newStep = rest;
-    }
-
-    const remindersFromStep = await ctx.prisma.workflowReminder.findMany({
-      where: {
-        workflowStepId: oldStep.id,
-      },
-      select: {
-        id: true,
-        referenceId: true,
-        method: true,
-        booking: {
-          select: {
-            eventTypeId: true,
-          },
-        },
-      },
-    });
-    //step was deleted
-    if (!newStep) {
-      // cancel all workflow reminders from deleted steps
-      await WorkflowRepository.deleteAllWorkflowReminders(remindersFromStep);
+  await Promise.all(
+    userWorkflow.steps.map(async (oldStep) => {
```

Comment:
Without Promise.all we were getting into some race conditions while deleting workflow with agent and phoneNumber